code/SingleRecordAdmin.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace SilverStripe\Admin;
+
+use SilverStripe\Forms\Form;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * The base class for admin sections that exist for editing a single record.
+ */
+abstract class SingleRecordAdmin extends LeftAndMain
+{
+    /**
+     * Determines if there should be a single record in the database that this admin edits.
+     *
+     * If this is not true, you need to provide a mechanism to tell the form which record it should be editing.
+     * This could be an action (e.g. edit/$ID), or could be based on something about the current member, etc.
+     */
+    private static bool $only_one_record = true;
```

Comment:
```suggestion
    private static bool $single_record = true;
```
`$only_one_record` sounds weird to me, almost implying that it's a bad thing :-)

Reply:
Feel free to suggest an alternative.

Reply:
$single_record

Reply:
How about `only_single_record`? Or `restrict_to_single_record`? I want the name to be fairly clear that it's saying "there should only ever be one record of this type" when set to true.

Reply:
$restrict_to_single_record sounds good

Reply:
Done